community/modules/scheduler/schedmd-slurm-gcp-v6-controller/modules/slurm_files/scripts/util.py

```diff
@@ -1928,26 +1932,58 @@ def nodeset_map(self, hostnames: list):
             nodeset_map[self.node_nodeset_name(node)].append(node)
         return nodeset_map
 
+    def _get_job_info(self, job_info, job_id=None) -> Optional[Job]:
+        """Extract job details"""
+        if job_id is None:
+            if match:= re.search(r"JobId=(\d+)", job_info):
```

Comment:
It's never the case that `JobId` is not present, should throw an exception if this happens.